### PR TITLE
CI: Rename workflow names

### PR DESCRIPTION
**Pull request by brlbil**
This commit changes the workflow's name fields to make them more human-readable by adding spaces.
Also, some of the names were changed to be more consistent with each other.

### CODE REVIEW

**Review by gentoo-root — APPROVED** [review]

File: .github/workflows/build-images-base.yaml
Comment (nbusseneau) [inline .github/workflows/build-images-base.yaml:26]: Do you think it's useful to have names on all jobs? I'd advocate for explicitly not setting names on jobs unless absolutely useful, and instead make sure the job IDs correctly represent what the job is doing. I think in general the less we need to worry about additional fields the better.

File: .github/workflows/conformance-clustermesh.yaml
Comment (nbusseneau) [inline .github/workflows/conformance-clustermesh.yaml:1]: If the trigger is changed then `.github/ariane-config.yaml` needs to be updated as well :)

File: .github/workflows/conformance-e2e.yaml
Comment (nbusseneau) [inline .github/workflows/conformance-e2e.yaml:1]: For any workflow name change, we will need to make sure PR required checks are updated accordingly. Once this PR is finalized and before it is merged, can we prepare the list so as to help maintainers?

**Review by nbusseneau — COMMENTED** [review]
Thanks, I've added some comments below on general strategy, can do another pass later on specifics.

File: .github/workflows/build-images-base.yaml
Comment (brlbil) [reply .github/workflows/build-images-base.yaml:26]: The idea behind giving these names is to make them more readable for new report tooling

File: .github/workflows/conformance-e2e.yaml
Comment (brlbil) [reply .github/workflows/conformance-e2e.yaml:1]: 👍

File: .github/workflows/conformance-clustermesh.yaml
Comment (brlbil) [reply .github/workflows/conformance-clustermesh.yaml:1]: I am not sure what do you mean with update, `.github/ariane-config.yaml` selects workflows by file name. We do not change the file names here.

Comment (nbusseneau) [reply .github/workflows/conformance-clustermesh.yaml:1]: In the change above, the trigger advertised in the workflow name went from `ci-multicluster` to `ci-clustermesh`, so the corresponding trigger needs to be updated in Ariane config.

File: .github/workflows/build-images-base.yaml
Comment (nbusseneau) [reply .github/workflows/build-images-base.yaml:26]: But is there really a big readability advantage in having jobs named `Build and Push Images` instead of `build-and-push`, as long as they have the same name everywhere? 🤔 I would advocate for not introducing bloat if we can avoid it, if everybody else is convinced this is sound then let's do it but I personally have no problem with having `build-and-push` as job name in the dashboard.

File: .github/workflows/conformance-clustermesh.yaml
Comment (brlbil) [reply .github/workflows/conformance-clustermesh.yaml:1]: I see, I'll fix it.

File: .github/workflows/build-images-base.yaml
Comment (brlbil) [reply .github/workflows/build-images-base.yaml:26]: I am not sure how much of a readability advantage. I have added a workflow with both jobs with names and without to see the difference. In my option, it is more pleasant if the names are separated by space.

Screenshot

File: .github/workflows/build-images-beta.yaml
Comment (nebril) [inline .github/workflows/build-images-beta.yaml:23]: Should this denote that these are beta images to differentiate from other types of images?

File: .github/workflows/build-images-ci.yaml
Comment (nebril) [inline .github/workflows/build-images-ci.yaml:38]: Should this denote that these are CI images to differentiate from other types of images?

File: .github/workflows/build-images-releases.yaml
Comment (nebril) [inline .github/workflows/build-images-releases.yaml:20]: Should this denote that these are release images to differentiate from other types of images?

**Review by nebril — APPROVED** [review]

File: .github/workflows/build-images-base.yaml
Comment (nbusseneau) [reply .github/workflows/build-images-base.yaml:26]: I guess the question is "is it pleasant enough to warrant additional bloat in workflows". Personally I would say no, but again I am only one person, would be nice to have more opinions :D

File: .github/workflows/build-images-beta.yaml
Comment (brlbil) [reply .github/workflows/build-images-beta.yaml:23]: This is a `job` name, the `workflow` name is already unique. It describes what the job does.
Same for the others.

File: .github/workflows/build-images-base.yaml
Comment (aanm) [reply .github/workflows/build-images-base.yaml:26]: @brlbil I don't understand your comment, should there be two pictures?

Comment (aanm) [reply .github/workflows/build-images-base.yaml:26]: Although 'name' is not mandatory, we should still add it to make it explicit and consistent across other workflows.

**Review by aanm — APPROVED** [review]

File: .github/workflows/build-images-base.yaml
Comment (nbusseneau) [reply .github/workflows/build-images-base.yaml:26]: Consistency I agree with -- but if there's already the same ID across workflows, then I'm personally not in favor of additionally copy/pasting the same name. Again, if I'm alone on this hill, let's just go with it :D

**Comment by brlbil** [comment]
Reviews in, failed tests are not related to the changes, setting as ready-to-merge

**Comment by joestringer** [comment]
@brlbil @nbusseneau how do we ensure consistency over time for these? I would like to avoid making rename changes like this as much as possible in future due to the disruption to PRs (renaming => have to rebase and rerun CI for every PR). Every time we create a new CI job there is a risk that we'll deviate again and then later on end up submitting a change again. We are still periodically adding more CI jobs. Is this something that @cilium/ci-structure will now enforce?

**Comment by nbusseneau** [comment]
I think yes, the right way forward would be to be stricter when doing reviews.

**Pull request merged**
